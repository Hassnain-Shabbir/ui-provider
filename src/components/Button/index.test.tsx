import React from "react"
import { render } from "@testing-library/react"
import {
  createRenderer as createShallowRenderer,
  ShallowRenderer,
} from "react-test-renderer/shallow"

import Component, { Props } from "./Button"

const requiredProps: Props = {}

describe("snapshot", () => {
  let renderer: ShallowRenderer

  beforeEach(() => {
    renderer = createShallowRenderer()
  })
  afterEach(() => {
    renderer.unmount()
  })

  it("required props only", () => {
    renderer.render(
      <Component {...requiredProps}>
        <div>Test</div>
      </Component>,
    )
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it("disabled", () => {
    const props: Props = {
      ...requiredProps,
      disabled: true,
    }
    renderer.render(
      <Component {...props}>
        <div>Test</div>
      </Component>,
    )
    expect(renderer.getRenderOutput()).toMatchSnapshot()

    const { container } = render(
      <Component {...props}>
        <div>Test</div>
      </Component>,
    )
    const button = container.querySelector(".disabled")
    expect(button).toBeDefined()
    expect(button).not.toBeNull()
  })
})
