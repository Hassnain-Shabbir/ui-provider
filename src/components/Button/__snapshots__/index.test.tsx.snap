// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot disabled 1`] = `
<div
  className="btn-wrapper disabled"
>
  <ForwardRef(Button)
    className="button disabled"
    disabled={true}
  >
    <div
      className="content"
    >
      <div>
        Test
      </div>
    </div>
  </ForwardRef(Button)>
</div>
`;

exports[`snapshot required props only 1`] = `
<div
  className="btn-wrapper"
>
  <ForwardRef(Button)
    className="button"
  >
    <div
      className="content"
    >
      <div>
        Test
      </div>
    </div>
  </ForwardRef(Button)>
</div>
`;
